server/webapp/WEB-INF/rails.new/app/presenters/api_v3/plugin/plugin_info_representer.rb

```diff
@@ -21,24 +21,77 @@ class PluginInfoRepresenter < BaseRepresenter
       REPRESENTER_FOR_PLUGIN_INFO_TYPE = {
         com.thoughtworks.go.plugin.domain.authentication.AuthenticationPluginInfo => AuthenticationPluginInfoRepresenter,
         com.thoughtworks.go.plugin.domain.authorization.AuthorizationPluginInfo => AuthorizationPluginInfoRepresenter,
-        com.thoughtworks.go.plugin.domain.notification.NotificationPluginInfo => NotificationPluginInfoRepresenter,
         com.thoughtworks.go.plugin.domain.packagematerial.PackageMaterialPluginInfo => PackageRepositoryPluginInfoRepresenter,
         com.thoughtworks.go.plugin.domain.pluggabletask.PluggableTaskPluginInfo => PluggableTaskPluginInfoRepresenter,
         com.thoughtworks.go.plugin.domain.scm.SCMPluginInfo => SCMPluginInfoRepresenter,
         com.thoughtworks.go.plugin.domain.elastic.ElasticAgentPluginInfo => ElasticPluginInfoRepresenter,
-        com.thoughtworks.go.plugin.domain.configrepo.ConfigRepoPluginInfo => ConfigRepoPluginInfoRepresenter,
-        com.thoughtworks.go.plugin.domain.common.BadPluginInfo => BadPluginInfoRepresenter,
       }
 
-      class <<self
-        def new(plugin)
-          representer_for_type(plugin.class).new(plugin)
-        end
+      alias_method :plugin, :represented
+
+      link :self do |opts|
+        opts[:url_builder].apiv3_admin_plugin_info_url(id)
+      end
+
+      link :doc do |opts|
+        'https://api.gocd.org/#plugin-info'
+      end
+
+      link :find do |opts|
+        opts[:url_builder].apiv3_admin_plugin_info_url(id: '__plugin_id__').gsub(/__plugin_id__/, ':plugin_id')
+      end
+
+      property :id, exec_context: :decorator
+      property :getExtensionName, as: :type
+      property :status, exec_context: :decorator do
+        property :state, getter: lambda {|*| state.to_s.downcase}
+        property :messages, if: lambda {|args| !self.messages.blank?}
+      end
+
+      property :plugin_file_location, exec_context: :decorator
+      property :isBundledPlugin, as: :bundled_plugin, exec_context: :decorator
 
-        def representer_for_type(type)
-          REPRESENTER_FOR_PLUGIN_INFO_TYPE[type] || (raise UnprocessableEntity, "Invalid plugin type '#{type}'. It has to be one of '#{REPRESENTER_FOR_PLUGIN_INFO_TYPE.keys.join(', ')}.'")
+      property :about, exec_context: :decorator do
+        property :name
+        property :version
+        property :target_go_version
+        property :description
+        property :target_operating_systems, getter: lambda {|opts| self.target_operating_systems.to_a}
+
+        property :vendor do
+          property :name
+          property :url
         end
       end
+
+      property :plugin_settings,
```

Comment:
I think even plugin_settings should go inside extension_info since all plugins do not have plugin_settings